### PR TITLE
Test utils

### LINKED ISSUE #1647: [RFC] Testing selectors API
- Browser integration testing requires some sort of attribute to identify elements in the DOM
- In the past many used `class="..."` but CSS-in-JS kills that
- Today many are manually adding some sort of `[data-test-id]` attribute
- But that's like... work... and... stuff
- Instead what if styled-components abstracted that away

For example, a component:

```js
import React from 'react';
import styled from 'styled-components';

export const ConfirmForm = styled.form`...`;
export const ConfirmButton = styled.button`...`;

export default function ConfirmPage(props) {
  return (
    <ConfirmForm onSubmit={props.onSubmit}>
      <ConfirmButton type="submit">Confirm.</ConfirmButton>
    </ConfirmForm>
  );
}
```

Then in your tests:

```js
import React from 'react';
import { render } from 'react-dom';
import ConfirmPage, { ConfirmForm, ConfirmButton } from './ConfirmPage';
import { find, findAll } from 'styled-components/test';

let root = document.getElementById('root');
render(<ConfirmPage/>, root);

let form = find(root, ConfirmForm); // <form/>
let buttons = findAll(form, ConfirmButton);  // [<button type="submit"/>]
```

You can actually write these methods today with styled-components:

```js
function find(el, component) {
  return el.querySelector("." + component.styledComponentId);
}

function findAll(el, component) {
  return el.querySelectorAll("." + component.styledComponentId);
}
```

See: https://codesandbox.io/s/1ryxvkkmrl

### PR DESCRIPTION
**Pull request by jamiebuilds**
Closes #1647

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,17 +1,21 @@
 {
   "name": "styled-components",
   "version": "3.2.5",
-  "description": "Visual primitives for the component age. Use the best bits of ES6 and CSS to style your apps without stress 💅",
+  "description":
+    "Visual primitives for the component age. Use the best bits of ES6 and CSS to style your apps without stress 💅",
```
Comment (jamiebuilds) [inline package.json]: Prettier decided to update your package.json

File: package.json [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
   "main": "dist/styled-components.cjs.js",
   "jsnext:main": "dist/styled-components.es.js",
   "module": "dist/styled-components.es.js",
   "react-native": "dist/styled-components.native.cjs.js",
   "browser": {
     "./dist/styled-components.es.js": "./dist/styled-components.browser.es.js",
-    "./dist/styled-components.cjs.js": "./dist/styled-components.browser.cjs.js",
-    "./dist/styled-components-no-parser.es.js": "./dist/styled-components-no-parser.browser.es.js",
-    "./dist/styled-components-no-parser.cjs.js": "./dist/styled-components-no-parser.browser.cjs.js"
+    "./dist/styled-components.cjs.js":
```
Comment (probablyup) [inline package.json]: hmm, this is definitely weird. would you mind reverting it and adding package.json to prettierignore?

**Comment by mxstbr** [comment]
Thank you so much for helping us improve styled-components! Based on our [Community Guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md) every person that has a PR of any kind merged is offered **an invitation to the styled-components organization**—that is you right now!

Should you accept, you'll get write access to the main repository. (and a fancy `styled-components` logo on your GitHub profile!) You'll be able to label and close issues, merge pull requests, create new branches, etc. We want you to help us out with those things, so don't be scared to do them! Make sure to read our [contribution and community guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md), which explains all of this in more detail. Of course, if you have any questions just let me know!

**Comment by jamiebuilds** [comment]
Oooof, @mxstbr this auto-invite thing is actually rather annoying

**Comment by mxstbr** [comment]
@jamiebuilds whaddaya mean? It should've caught that you're already a member, that's a bug :confused:

**Comment by jamiebuilds** [comment]
It subscribed me automatically and sent a whole bunch of notifications

**Comment by mxstbr** [comment]
I guess GitHub didn't send you a proper invitation because you were already a member of the org, so they just added you to that team. Normally you'd get an invite you have to accept before anything happens—you're the absolute edge case. 😅

**Pull request merged**